### PR TITLE
Add --public option for Stub deployments

### PR DESCRIPTION
**Pull request by gongy**

### CODE REVIEW

**Comment by RohanArepally** [comment]
@gongy just curious what's this for?

**Comment by gongy** [comment]
@RohanArepally published apps will have functions which other workspaces can look up and derive from.

**Review by aksh-at — APPROVED** [review]
LGTM. We might have to deprecate this in favor of `visibility="public"` at some point, if we actually add other visibility levels like the internal enum allows for. But that's not a high likelihood any time soon so this interface is better for now.

File: modal/cli/run.py [hunk 1]
```diff
@@ -262,7 +265,15 @@ def deploy(
     if name is None:
         name = stub.name
 
-    deploy_stub(stub, name=name, environment_name=env)
+    if public:
+        if not click.confirm(
+            "⚠️ Public apps are a beta feature.⚠️\n"
+            "Deploying an app as public means any user can lookup and use its functions.\n"
```
[the remark this answers is not in the document]
Comment (gongy) [reply modal/cli/run.py:271]: Thanks, needed this copy review 👍

[the remark this answers is not in the document]
Comment (gongy) [reply modal/cli/run.py:271]: Going to full send the visibility enum right now. It's free future-proofing

**Review by aksh-at — APPROVED** [review]

**Pull request merged**